### ISSUE TITLE
Pillow cannot find libjpeg.so.9 (even though libjpeg.so.8 exists)

### ISSUE THREAD

**Issue by AbdealiJK** [python-pillow/Pillow#1952]
### What did you do?
I installed Pillow from `pip`. It installed version 3.2.0. Then when i ran some code which works on another system. I further on found `python -c "from PIL import _imaging"` does not work.

### What did you expect to happen?
I was expecting it to work fine, works fine in another Ubuntu 14.04 with conda, and pretty much the same setup.

### What actually happened?
```
$ python -c "from PIL import _imaging"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: libjpeg.so.9: cannot open shared object file: No such file or directory
```

I figured maybe libjpeg isn't Installed, but libjpeg is installed in apt-get:
```
$ dpkg --get-selections | grep libjpeg
libjpeg-dev:amd64                               install
libjpeg-turbo-progs                             install
libjpeg-turbo8:amd64                            install
libjpeg-turbo8-dev:amd64                        install
libjpeg8:amd64                                  install
libjpeg8-dev:amd64                              install
```

Then I thought maybe ldconfig wasnt set up correctly, but it was able to find the file:
```
$ ldconfig -p | grep -i libjp
        libjpeg.so.8 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libjpeg.so.8
        libjpeg.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libjpeg.so
```

I even tried installing it in the conda environment, but it still gave the same error:
```
$ conda install jpeg
$ ls ~/miniconda/envs/py27/lib/libjpeg.so*
~/miniconda/envs/py27/lib/libjpeg.so
~/miniconda/envs/py27/lib/libjpeg.so.8
~/miniconda/envs/py27/lib/libjpeg.so.8.4.0
```
### What versions of Pillow and Python are you using?
Pillow v3.2.0
Python v2.7.11 :: Continuum Analytics
pip v8.1.2
conda v4.0.7
Ubuntu 14.04

**Comment by AbdealiJK** [comment]
I even tried:
```
$ ln -s ~/miniconda/envs/py27/lib/libjpeg.so.8.4.0 ~/miniconda/envs/py27/lib/libjpeg.so.9
```
And that gave:
```
$ python -c "from PIL import _imaging"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: /data/project/catimages/miniconda/envs/noconda/bin/../lib/libjpeg.so.9: version `LIBJPEG_9.0' not found (required by /data/project/catimages/miniconda/envs/noconda/lib/python2.7/site-packages/PIL/_imaging.so)
```

It seems like libjpeg 9 has now become the minimum version, but for some reason it still works with libjpeg8 in another system with the exact same setup with exact same versions of everything.

**Comment by wiredfool** [comment]
Pillow needs the same version of the libraries as it was built with. So, when it was build against libjpeg.so.9, it's going to require that in the future. Your best bet is to chose one set of jpeg libraries + dev headers and rebuild pillow against those. 

Conda makes it a bit more complicated, since they preload some of their libraries and Pillow has some trouble determining that at build time.

**Comment by AbdealiJK** [comment]
@wiredfool Thanks for the reply :)

I compiled Pillow (With `pip install Pillow`) on the same machine. And there are no traces of libjpeg9 on the system that I am able to find (`ldconfig` does not detect it either).

I followed up your suggestion and did a uninstall of Pillow and a reinstall, and it now seems to work.
I'm not sure what the cause could have been ...

Closing this issue as my problem seems to be solved with a reinstall.
